src/overmap.cpp

```diff
@@ -2324,38 +2324,42 @@ void overmap::draw(WINDOW *w, WINDOW *wbar, const tripoint &center,
         }
     }
     if (has_target && blink &&
-        (target.x < cursx - om_half_width ||
-         target.x >= cursx + om_half_width  ||
-         target.y < cursy - om_half_height ||
-         target.y >= cursy + om_half_height)) {
+        (target.x < offset_x ||
+         target.x >= offset_x + om_map_width ||
+         target.y < offset_y ||
+         target.y >= offset_y + om_map_height)) {
+        int marker_x = ( target.x >= offset_x ) ? ( ( target.x < offset_x + om_map_width  ) ? target.x - offset_x : om_map_width  - 1) : 0;
```

Comment:
Nested ternary expressions make the code hard to read, please break this into a series of if statements, or multiple variables e.g. 'above_window', 'below_window', etc